collectors/python.d.plugin/vnstat/vnstat.chart.py

```diff
@@ -0,0 +1,731 @@
+                            new_line
+                        )
+
+                        self.add_chart_dimension("years_total_" + interface_or_chart, new_line)
+
+                        data[
+                            (
+                                "years_total_"
+                                + interface
+                                + "_year"
+                                + str(hash(year[0]))[1:]
+                            )
+                        ] = year[1]
+
+            charts["totals_" + interface_or_chart] = {
+                "options": [
+                    None,
+                    "Total RX vs. Total TX",
+                    "B",
+                    "total bandwidth " + interface,
+                    "vnstat.totals_" + interface_or_chart,
+                    "stacked",
+                ],
+                "lines": [
+                    ["totals_" + interface + "_rx", "rx", "absolute", 1, 1],
+                    ["totals_" + interface + "_tx", "tx", "absolute", 1, 1],
+                ],
+            }
+
+            order.append("totals_" + interface_or_chart)
+
+            data["totals_" + interface + "_rx"] = data_points[index][0]
+            data["totals_" + interface + "_tx"] = data_points[index][1]
+
+            index += 1
+
+        return charts, order, data
+
+    def _get_data(self):
+        """
+        Parse vnstat command output
+        :return: dict
+        """
+
+        raw_output = str(self._get_raw_data()[0]).strip()
+        output = json.loads(raw_output)
+
+        # Check for version compatibility and return None if incompatible:
+        try:
+            if (not int(output['jsonversion']) >= 2):
+                return
+        except ValueError:
+            return
+
+        allowed_interfaces = self.configuration.get("interface", "all").split()
+
+        interfaces = []
+        data_points = []
+        for interface in output["interfaces"]:
+            if (not ("all" in allowed_interfaces)) and (
+                not (interface["name"] in allowed_interfaces)
+            ):
+                continue
+
+            interfaces.append(interface["name"])
+
+            traffic = interface["traffic"]
+
+            total_rx = traffic["total"]["rx"]
+            total_tx = traffic["total"]["tx"]
+
+            current_amount = (
+                traffic["fiveminute"][(len(traffic["fiveminute"]) - 1)]["rx"]
+                + traffic["fiveminute"][(len(traffic["fiveminute"]) - 1)]["tx"]
+            )
+            current_data_rate = ((current_amount * 8) / 1000) / 300
+            # If no data is present for yesterday, then today's values will be chosen automatically:
+            yesterdayAmount = (
+                traffic["day"][(len(traffic["day"]) - 2)]["rx"]
+                + traffic["day"][(len(traffic["day"]) - 2)]["tx"]
+            )
+            yesterday_data_rate = ((yesterdayAmount * 8) / 1000) / 86400
```

Comment:
is `yesterdayAmount` float? If no wee to make at least one number float

this is py2 problem and it is deprecated, but still
```python
>>> 10/3
3
```